packages/transport/test/modules/relationships/TerminateDecompose.test.ts

```diff
@@ -0,0 +1,49 @@
+import { IDatabaseConnection } from "@js-soft/docdb-access-abstractions";
+import { AccountController, CoreId, RelationshipChangedEvent, Transport } from "../../../src";
+import { TestUtil } from "../../testHelpers/TestUtil";
+
+describe("Terminate and Decompose simultaneously", function () {
+    let connection: IDatabaseConnection;
+
+    let transport: Transport;
+
+    let sender: AccountController;
+    let recipient: AccountController;
+
+    let relationshipId: CoreId;
+
+    beforeAll(async function () {
+        connection = await TestUtil.createDatabaseConnection();
+        transport = TestUtil.createTransport(connection);
+
+        await transport.init();
+
+        const accounts = await TestUtil.provideAccounts(transport, 2);
+        sender = accounts[0];
+        recipient = accounts[1];
+
+        const relationships = await TestUtil.addRelationship(sender, recipient);
+        relationshipId = relationships.acceptedRelationshipFromSelf.id;
+    });
+
+    afterAll(async function () {
+        await sender.close();
+        await recipient.close();
+        await connection.close();
+    });
+
+    test("should only get one RelationshipChangedEvent for Terminate and Decompose at the same time", async function () {
+        await sender.relationships.terminate(relationshipId);
+        await sender.relationships.decompose(relationshipId);
+
+        const events: RelationshipChangedEvent[] = [];
+        transport.eventBus.subscribe(RelationshipChangedEvent, function (event) {
+            events.push(event);
+        });
```

Comment:
Is the order of initiating events and subscribing to then irrelevant?

Reply:
what?

Reply:
I have to subscribe to events BEFORE they will be triggered, so yeah - the order is very relevant :D

Reply:
Ah and the relevant call is the sync and not the terminate/decompose, right?

Reply:
all (good) tests have the same setup `prepare - act - validate`, this is done exactly like this.

Reply:
So that's a yes? The events that are subscribed to arrive when the sync is done and not when the sender terminates the Relationship, don't they?

Reply:
I think so, yes?

Reply:
According to your first answer, this should be the case. Otherwise we wouldn't receive the events. I was just wondering why you subscribed to the event bus **after** the relationship is terminated and decomposed.

Reply:
I am so confused of this conversation :D

Reply:
Ok, never mind. The tests are green so everything seems to be working as it should. :D